cabot/cabotapp/models.py

```diff
@@ -934,6 +940,37 @@ def _run(self):
         return result
 
 
+class TCPStatusCheck(StatusCheck):
+
+    @property
+    def check_category(self):
+        return "TCP Check"
+
+    @property
+    def description(self):
+        return 'Monitoring connection of {0}:{1}'.format(self.address, self.port)
+
+    address = models.CharField(max_length=1024, help_text='IP address or hostname to monitor',)
+    port = models.PositiveIntegerField(help_text='Port to listen on',)
+    timeout = models.PositiveIntegerField(default='8', help_text='Time to wait on idle connection',)
+
+    update_url = 'update-tcp-check'
+
+    icon = 'glyphicon glyphicon-text-wide'
+
+    def _run(self):
+        result = StatusCheckResult(check=self)
+
+        try:
+            socket.create_connection((self.address,self.port), self.timeout)
```

Comment:
Should we enforce a maximum timeout? To prevent people from adding a bunch of checks with super high timeouts and blocking other checks from running (this could also be an issue with HTTPStatusChecks)